torch/csrc/distributed/rpc/request_callback_no_python.cpp

```diff
@@ -420,18 +414,20 @@ void RequestCallbackNoPython::processForwardAutogradReq(
 
         if (wrappedRpcResponseFuture.hasError()) {
           // Propagate error to responseFuture if we had one.
-          responseFuture->setError(wrappedRpcResponseFuture.exception_ptr());
+          std::rethrow_exception(wrappedRpcResponseFuture.exception_ptr());
```

Comment:
IIUC, Future::then wraps this callback, captures this exception, and sets the error on the returned child Future?

Reply:
Correct: the beauty of `then` is that we can almost forget about the error path: `then` will capture and store any exception for us automatically, and it's impossible to mess it up. In addition, calling `value` on the parent future will automatically rethrow the exception if any. Hence consider this code: `childFut.then([](Future& parentFut) { return doSth(parentFut.value()); });`. This code does correct error propagation, without even thinking about it.